tests/trx_finality_status_test.py

```diff
@@ -0,0 +1,243 @@
+    irreversibleState = "IRREVERSIBLE"
+    unknownState = "UNKNOWN"
+    status = []
+    state = None
+    i = 0
+    preInfo = None
+    postInfo = None
+    transferAmount=10
+
+    # ensuring that prod0's producer is active, which will give sufficient time to identify the transaction as "LOCALLY_APPLIED" before it travels
+    # through the chain of nodes to nod0 to be added to a block
+    # defproducera -> defproducerb -> defproducerc -> NPN
+    prod0.waitForProducer("defproducera", exitOnError=True)
+    testNode.transferFunds(cluster.eosioAccount, account1, "{}.0000 {}".format(transferAmount, CORE_SYMBOL), "fund account")
+    transId=testNode.getLastSentTransactionId()
+    retStatus=testNode.getTransactionStatus(transId)
+    state = getState(retStatus)
+
+    assert state == localState, \
+        Print("ERROR: getTransactionStatus didn't return \"{}\" state.\n\nstatus: {}".format(localState, json.dumps(retStatus, indent=1)))
+    status.append(copy.copy(retStatus))
+    startingBlockNum=testNode.getInfo()["head_block_num"]
+
+    def validateTrxState(status, present):
+        bnPresent = "block_number" in status
+        biPresent = "block_id" in status
+        btPresent = "block_timestamp" in status
+        desc = "" if present else "not "
+        group = bnPresent and biPresent and btPresent if present else not bnPresent and not biPresent and not btPresent
+        assert group, \
+            Print("ERROR: getTransactionStatus should{} contain \"block_number\", \"block_id\", or \"block_timestamp\" since state was \"{}\".\nstatus: {}".
+                  format(desc, getState(status), json.dumps(status, indent=1)))
+
+    validateTrxState(status[0], present=False)
+
+    def validate(status, knownTrx=True):
+        assert "head_number" in status and "head_id" in status and "head_timestamp" in status, \
+            Print("ERROR: getTransactionStatus should contain \"head_number\", \"head_id\", and \"head_timestamp\" always.\nstatus: {}".
+                  format(json.dumps(status, indent=1)))
+
+        assert "irreversible_number" in status and "irreversible_id" in status and "irreversible_timestamp" in status, \
+            Print("ERROR: getTransactionStatus should contain \"irreversible_number\", \"irreversible_id\", and \"irreversible_timestamp\" always.\nstatus: {}".
+                  format(json.dumps(status, indent=1)))
+
+        assert not knownTrx or "expiration" in status, \
+            Print("ERROR: getTransactionStatus should contain \"expiration\" always.\nstatus: {}".
+                  format(json.dumps(status, indent=1)))
+
+        assert "earliest_tracked_block_id" in status and "earliest_tracked_block_number" in status, \
+            Print("ERROR: getTransactionStatus should contain \"earliest_tracked_block_id\" and \"earliest_tracked_block_number\" always.\nstatus: {}".
+                  format(json.dumps(status, indent=1)))
+
+    validate(status[0])
+
+    numTries = 5
```

Comment:
Why not Node `waitForTransInBlock()` with a timeout of 5 * block time?  Or better yet 12 * block time, to guarantee the current producer finishes its turn and the next producer produces at least one block.

Generally speaking, when picking arbitrary delays, be cognizant of block times and the duration of a producer's turn (when there's more than one) to give the system the best chance to complete the test successfully (when not testing expected behavior which is fully deterministic).